File: alwin-middleware-grapescode/alwin-core-api/src/main/java/com/codersteam/alwin/core/api/service/notification/NotificationService.java
package com.codersteam.alwin.core.api.service.notification;

import com.codersteam.alwin.core.api.notification.NotificationDto;
import com.codersteam.alwin.core.api.notification.NotificationsWithCountsDto;

import javax.ejb.Local;

/**
 * Serwis do obsługi powiadomień
 *
 * @author Adam Stepnowski
 */
@Local
public interface NotificationService {

    /**
     * Pobieranie powiadomień po identyfikatorze zlecenia
     *
     * @param issueId -  identyfikator zlecenia
     * @return powiadomienia wraz z informacją o ich łącznej liczbie oraz ilości nieprzeczytanych
     */
    NotificationsWithCountsDto findAllIssueNotifications(final Long issueId);

    /**
     * Tworzenie nowego powiadomienia
     *
     * @param notificationDto - dane powiadomienia
     */
    void createNotification(NotificationDto notificationDto);

    /**
     * Pobieranie powiadomienia po identyfikatorze
     *
     * @param notificationid - identyfikator powiadomienia
     * @return powiadomienie
     */
    NotificationDto findNotification(final Long notificationid);

    /**
     * Oznacza powiadomienie jako przeczytane, poprzez aktualizację daty odczytania
     *
     * @param notificationId - identyfikator powiadomienia
     */
    void markNotificationAsRead(Long notificationId);
}

File: alwin-web-grapescode/src/app/notification/notification.service.ts
import {Injectable} from '@angular/core';
import {AlwinHttpService} from '../shared/authentication/alwin-http.service';
import {Observable} from 'rxjs';
import {NotificationsWithCounts} from './notifications-with-counts';
import {Notification} from './notification';

/**
 * Serwis dostępu do powiadomień
 */
@Injectable()
export class NotificationService {
  constructor(private http: AlwinHttpService) {
  }

  /**
   * Pobiera wszystkie powiadomienia dla podanego zlecenia wraz z ilością nieprzeczytanych oraz wszystkich powiadomień
   * @param {number} issueId - identyfikator zlecenia
   *
   * @returns {Observable<NotificationsWithCounts>} - lista powiadomień z licznikami
   */
  getAllNotificationsWithCounters(issueId: number): Observable<NotificationsWithCounts> {
    return this.http.get<NotificationsWithCounts>(`notifications/issue/${issueId}`);
  }

  /**
   * Oznacza powiadomienie jako przeczytane
   * @param {Notification} notification - powiadomienie do przeczytania
   *
   * @returns {any} - poprawny kod aktualizacji powiadomienia 202
   */
  updateNotificationReadDate(notification: Notification): any {
    return this.http.patch(`notifications/${notification.id}`, notification);
  }

  /**
   * Tworzy nowe powiadomienie
   * @param {Notification} notification - powiadomienie do utworzenia
   *
   * @returns {any} - poprawny kod utworzenia powiadomienia 201
   */
  createNotification(notification: Notification): any {
    return this.http.postObserve('notifications/', notification);
  }
}
